rosbag2_storage_default_plugins/include/rosbag2_storage_default_plugins/sqlite/sqlite_storage.hpp

```diff
@@ -92,6 +93,7 @@ class ROSBAG2_STORAGE_DEFAULT_PLUGINS_PUBLIC SqliteStorage
   void fill_topics_and_types();
   void activate_transaction();
   void commit_transaction();
+  void write_locked(std::shared_ptr<const rosbag2_storage::SerializedBagMessage> message);
```

Comment:
```suggestion
  void write_locked(std::shared_ptr<const rosbag2_storage::SerializedBagMessage> message) REQUIRES(write_mutex_);
```

Reply:
Done - thank you for suggestions! I didn't know we had such annotations. 
I also rebased this PR on the new master.